osu.Server.BeatmapSubmission/Program.cs

```diff
@@ -81,6 +84,24 @@ public static void Main(string[] args)
                 }
             }
 
+            builder.Services.AddRateLimiter(options =>
+            {
+                options.RejectionStatusCode = (int)HttpStatusCode.TooManyRequests;
+                options.AddPolicy(RATE_LIMIT_POLICY,
+                    ctx =>
+                    {
+                        uint userId = ctx.User.GetUserId();
+
+                        return RateLimitPartition.GetSlidingWindowLimiter(userId, _ => new SlidingWindowRateLimiterOptions
+                        {
+                            PermitLimit = 30,
```

Comment:
If anything, I'd agree at this being too lenient, and it could be dropped as low as 5 per minute. Maybe we start stringent and wait for issues to occur (because let's be honest: the most likely case is going to be a runaway client stuck in an infinite retry death spiral as opposed to someone trying to abuse the service).

```suggestion
                            PermitLimit = 5,
```

Reply:
Sure, although I've done a bit of a different change, see https://github.com/ppy/osu-server-beatmap-submission/pull/16/commits/592b03524d90b613e299e2b7e2c8a2c2f4112e0b. The rate limit is now 6 req/min with 2 requests being replenished to the pool every 20sec.

The reason why I want that specific sort of thing is that a full upload flow takes 2 requests (the initial one to set up the beatmaps and get the file listing, and the second one to either do a full .osz upload or a differential update). So I want these rate limits to not be odd numbers, to prevent half-failing submissions because the client ran out of ratelimit between one request and the other.

Reply:
Sounds fine, yeah.